javalib/src/main/scala/java/math/BigDecimal.scala

```diff
@@ -0,0 +1,1729 @@
+      // since -Integer.MIN_VALUE == Integer.MIN_VALUE
+      if (BitLength < 64) valueOf(SmallValue, safeLongToInt(newScale))
+      else new BigDecimal(getUnscaledValue, safeLongToInt(newScale))
+    } else if (-newScale < LongTenPows.length && BitLength + lptbLen < 64) {
+      valueOf(SmallValue * LongTenPows(-newScale.toInt), 0)
+    } else {
+      new BigDecimal(multiplyByTenPow(getUnscaledValue, safeLongToInt(-newScale)), 0)
+    }
+  }
+
+  /** Rounds for numbers which unscaled value fits in the type {@code long}.
+    *
+    * This method implements an efficient rounding for numbers which unscaled
+    * value fits in the type {@code long}.
+    *
+    * @param mc the context to use
+    * @param discardedPrecision the number of decimal digits that are discarded
+    * @see #round(MathContext)
+    */
+  private def smallRound(mc: MathContext, discardedPrecision: Int) {
+    val sizeOfFraction: Long = LongTenPows(discardedPrecision)
+    val newScale0: Long = Scale.toLong - discardedPrecision
+    val unscaledVal: Long = SmallValue
+    // Getting the integer part and the discarded fraction
+    val intPart0: Long = unscaledVal / sizeOfFraction
+    val fraction: Long = unscaledVal % sizeOfFraction
+    // If the discarded fraction is non-zero perform rounding
+    val (newScale: Long, intPart: Long) = {
+     if (fraction != 0) {
+       // To check if the discarded fraction >= 0.5
+       val compRem = longCompareTo(Math.abs(fraction) * 2, sizeOfFraction)
+       // To look if there is a carry
+       val frac = java.lang.Long.signum(fraction) * (5 + compRem)
+       val intPart1 = intPart0 + roundingBehavior(intPart0.toInt & 1, frac, mc.roundingMode)
+       // If after to add the increment the precision changed, we normalize the size
+       if (Math.log10(Math.abs(intPart1)) >= mc.precision)
+         (newScale0 - 1, intPart1 / 10)
+       else
+         (newScale0, intPart1)
+     } else {
+       (newScale0, intPart0)
+     }
+
+   }
+    // To update all internal fields
+    Scale = safeLongToInt(newScale)
+    Precision = mc.precision
+    SmallValue = intPart
+    BitLength = bitLength(intPart)
+    IntVal = null
+  }
+
+  /** Returns an exact value or throws an exception.
+    *
+    * If {@code intVal} has a fractional part throws an exception,
+    * otherwise it counts the number of bits of value and checks if it's out of
+    * the range of the primitive type. If the number fits in the primitive type
+    * returns this number as {@code long}, otherwise throws an
+    * exception.
+    *
+    * @param bitLengthOfType number of bits of the type whose value will be
+    *                        calculated exactly
+    * @return the exact value of the integer part of {@code BigDecimal}
+    *         when is possible
+    * @throws ArithmeticException when rounding is necessary or the
+    *         number don't fit in the primitive type
+    */
```

Comment:
The way you place the starts. It should be:
```scala
/** Text.
 *
 *  More text.
 */
```
instead of
```scala
/** Text.
  *
  * More text.
  */
```